inc/datahandlers/user.php

```diff
@@ -1783,4 +1783,61 @@ function clear_profile($delete_uids=false, $gid=0)
 			remove_avatars($uid);
 		}
 	}
+
+	public function verify_signature()
+	{
+		global $mybb, $parser;
+
+		$parser_options = array(
+			'allow_html' => $mybb->settings['sightml'],
+			'filter_badwords' => 1,
+			'allow_mycode' => $mybb->settings['sigmycode'],
+			'allow_smilies' => $mybb->settings['sigsmilies'],
+			'allow_imgcode' => $mybb->settings['sigimgcode'],
+			"filter_badwords" => 1
+		);
+
+		if($this->data['showimages'] != 1 && $this->data['uid'] != 0)
+		{
+			$parser_options['allow_imgcode'] = 0;
+		}
+
+		$parsed_sig = $parser->parse_message($this->data['signature'], $parser_options);
+
+		if((($mybb->settings['sigimgcode'] == 0 && $mybb->settings['sigsmilies'] != 1) &&
+			substr_count($parsed_sig, "<img") > 0) ||
+			(($mybb->settings['sigimgcode'] == 1 || $mybb->settings['sigsmilies'] == 1) &&
+			substr_count($parsed_sig, "<img") > $mybb->settings['maxsigimages'])
+		)
+		{
+			$imgsallowed = ($mybb->settings['sigimgcode'] == 1 ? $mybb->settings['maxsigimages'] : 0);
+			$this->set_error('too_many_sig_images2', array($imgsallowed));
+			$mybb->input['preview'] = 1;
```

Comment:
Preview handling definitely shouldn't be here. The line is useless anyways since preview is not available if there are any errors: https://github.com/dequeues/mybb/blob/fix2298/usercp.php#L1903